source/fab/artefacts.py

```diff
@@ -158,3 +157,18 @@ def __call__(self, artefact_store):
             build_lists[root] = filter_source_tree(source_tree=tree, suffixes=self.suffixes)
 
         return build_lists
+
+
+class ArtefactStore(dict):
```

Comment:
Is there a way to capture the type information, the `str, Any` through the inheritance? There's `TypeDict` but it is not supported in Python3.7 which we need to continue supporting for a while longer.

Reply:
I did some googling, the only option I found was to install an extension (and then import TypedDict depending on python version), but I feel that's rather messy.

The only thing I can think of would be to do:
```
    def __getitem__(self, key: str) -> Any:
        return super().__getitem__(key)
```

And I guess I could do the same for `get(...)`? 

Unfortunately (see above), my understanding of Python typing is not really up to decide what to do here :)